### PR TITLE
Add minimal dynamic page mode configuration class.

### PR DESCRIPTION
**Pull request by philbogle**
This class will let us switch between (e.g.) two different partner takeovers by setting a dynamic configuration variable, or apply more sophisticated A-B policies down the road.

There is a page_mode helper that is convenient to call from views. For example, in HAML:

```
  - if page_mode == 'starwars'
    <h3> starwars </h3>
  - if page_mode == 'minecraft'
    <h3> minecraft </h3>
```

### CODE REVIEW

File: lib/dynamic_config/page_mode.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  # Sets the default page mode for all requests.
+  # @param {string} page_mode
+  def set_default(page_mode)
+    DCDO.set('page_mode', page_mode)
+  end
+end
+
+PageMode = PageModeBase.new
```
Comment (philbogle) [inline lib/dynamic_config/page_mode.rb]: Fixed

**Comment by philbogle** [comment]
@trevorb @breville  FYI.

**Comment by caleybrock** [comment]
So we'll do something like this? Can we have multiple of these blocks on the same page?

if PageMode.get() == 'starwars' {
   showStarWarsContent();
} else {
   showOtherContent();
}

Looks good to me but @breville will know better if this will work with some of our gsheet generated content.

File: lib/dynamic_config/page_mode.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# A class for returning a "page mode" use by views to decided how to render
```
Comment (breville) [inline lib/dynamic_config/page_mode.rb]: decide*

**Comment by breville** [comment]
lgtm.. will have more feedback once we start using it :)  this should work fine with gsheet content, since we always pull that data down via server-side code in pegasus and can key off of it as we like.

File: lib/dynamic_config/page_mode.rb (hunk 2 above)
Comment (philbogle) [inline lib/dynamic_config/page_mode.rb]: Fixed

**Comment by philbogle** [comment]
I added a page_mode convenience helper you can call from views.

**Comment by caleybrock** [comment]
:+1: convenience helper is great lgtm

**Comment by breville** [comment]
@philbogle i don't see page_mode available on pegasus pages.  is there something extra needed?

**Pull request merged**